metagpt/utils/di_graph_repository.py

```diff
@@ -19,20 +21,87 @@
 
 
 class DiGraphRepository(GraphRepository):
+    """Graph repository based on DiGraph.
+
+    This class represents a graph repository that utilizes a directed graph (DiGraph) to manage relationships
+    between entities. It inherits from the GraphRepository class, providing a common interface for graph repositories.
+
+    Attributes:
+        _repo (DiGraph): The underlying directed graph representing the repository.
+
+    Methods:
+        insert: Insert a new triple into the graph repository.
+        select: Retrieve triples from the graph repository based on specified criteria.
+        delete: Delete triples from the graph repository based on specified criteria.
+        save: Save any changes made to the graph repository.
+        name: Get the name of the graph repository.
+
+    Example:
+        di_graph_repo = DiGraphRepository(name="MyDiGraphRepo")
+        di_graph_repo.insert(subject="Node1", predicate="connects_to", object_="Node2")
+        # Represents a directed relationship: Node1 connects_to Node2
+
+    Note:
+        This class extends the GraphRepository class and is specifically designed for managing directed relationships
+        using a DiGraph.
```

Comment:
Notes 24-46 are almost meaningless. After reading it, I still don't understand, what is the difference between it and GraphRepository? Their annotations seem to be exactly the same?